### ISSUE TITLE
Playlist minor accessibility improvements

### ISSUE THREAD

**Issue by petemill** [brave/brave-browser#32600]
1. Cannot change playlist item with keyboard. Every item that has a click handler should be able to be TABed to via the keyboard.
2. All buttons which don't have a text label should have a aria attribute label or a title attribute label. This would show via the screen reader, or via mouse hover (or keyboard focus).

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.61.61 Chromium: 119.0.6045.105 (Official Build) beta (64-bit)
-- | --
Revision | 6bbedb0b809f83b08334ffa36b1ea9df92e86245
OS | Windows 11 Version 22H2 (Build 22621.2428)
```

1. Installed 1.61.61
2. launched Brave
3. opened `brave://flags`
4. enabled `#playlist`
5. visited youtube.com
6. clicked on playlist button in the sidebar
7. verified Playlist UI opened in the sidepanel
8. pressed 'Tab' key 
9. confirmed `+` element in the web ui focused.
10. pressed keyboard `Enter` 
11. `New playlist` dialog shown 
12. entered `test2` and pressed tab and pressed `Enter`
13. confirmed `test2` playlist is created
14. pressed `Tab` again to move to next element 
15. confirmed `settings` icon is focused
16. pressed `keyboard` `Enter`
17. Nothing happened as the mouse action didn't work either.
18. confirmed `Play later` button is focused
19. added an item to the playlist `test`
20. able to successfully tab thru `test` playlist UI



https://github.com/brave/brave-browser/assets/98358127/4eda229a-0ff0-426b-983a-5fb072d19602

https://github.com/brave/brave-browser/assets/98358127/7d516298-4999-43c9-bb63-8b0cd10fe5c1
